client/js/editmode.js

```diff
@@ -295,6 +295,40 @@ function applyEditOptionsPiece(widget) {
   widget.color = $('#pieceColor').value;
 }
 
+//seat functions
+function populateEditOptionsSeat(widget) {
+  $('#seatPlayerColor').value = widget.color || "black";
+  $('#seatPlayerName').value = widget.player || "~ empty seat ~";
+  $('#seatEmpty').checked = false;
+}
+
+function applyEditOptionsSeat(widget) {
+  if ($('#seatEmpty').checked){
+    delete widget.player;
+    delete widget.color;
+  }
+  else if($('#seatPlayerName').value!="~ empty seat ~"){
+    toServer('playerColor', { player: widget.player, color: $('#seatPlayerColor').value });
+    toServer('rename', { oldName: widget.player, newName: $('#seatPlayerName').value });
+    widget.player = $('#seatPlayerName').value;
+    widget.color = $('#seatPlayerColor').value;
+  }
+}
+
+//spinner functions
+function populateEditOptionsSpinner(widget) {
+  }
+
+function applyEditOptionsSpinner(widget) {
+  for(let i=0; i<9; ++i) {
+    if($a('#spinnerOptions > [name=spinnerOptions]')[i].selected){
+      widget.options = JSON.parse($a('#spinnerOptions > [name=spinnerOptions]')[i].value);
+      delete widget.angle;
+      widget.value=widget.options[widget.options.length-1];
+    }
+  }
+}
+
```

Comment:
Perhaps I'm just blind, but I don't see a seat widget on the basic editor screen.

Reply:
Seems fair, but once #817 is merged, I don't think there will be a way to create a seat widget any more without starting from a basic widget.

Reply:
I prefer not to lock the PR behind this. There will be a composed widget for compatibility with PCIO but we haven't figured out yet what is the desired seat for VTT. I was hoping the discord group would come forward with ideas but no one answered my requests for help.

Reply:
> Seems fair, but once #817 is merged, I don't think there will be a way to create a seat widget any more without starting from a basic widget.

oh yeah I see the problem but anyways I don't want to lock the PR